### PR TITLE
Read access to read access k8s-artifacts-gcslogs

### PR DESCRIPTION
**Pull request by ameukam**
Ref: https://github.com/kubernetes/k8s.io/issues/1945

Initial request:
https://groups.google.com/g/kubernetes-wg-k8s-infra/c/Wkw0uyMKSXk/m/QLVIAMZzAAAJ.

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

**Comment by ameukam** [comment]
/hold
Need approval from chairs and TLs.
/assign @dims @spiffxp @thockin 
cc @hh @Riaankl

**Comment by hh** [comment]
Thanks @ameukam

File: infra/gcp/ensure-prod-storage.sh
Comment (dims) [inline infra/gcp/ensure-prod-storage.sh:362]: typo @ameukam ... `k8s-artifacts-gcslogs`

Comment (spiffxp) [reply infra/gcp/ensure-prod-storage.sh:362]:
```suggestion
    # Special case: empower II Coop folks to have read access to GCS bucket k8s-artifacts-gcs-logs
```

**Review by spiffxp — COMMENTED** [review]
One nit otherwise LGTM

**Comment by spiffxp** [comment]
I still feel like moving the bucket to its own project might better accomplish the access rules laid out in https://github.com/kubernetes/k8s.io/issues/904#issuecomment-816484483 but if this gets us started for now so be it

File: infra/gcp/ensure-prod-storage.sh
Comment (hh) [reply infra/gcp/ensure-prod-storage.sh:362]: 👍

Comment (ameukam) [reply infra/gcp/ensure-prod-storage.sh]: @spiffxp @dims Fixed.

File: infra/gcp/ensure-prod-storage.sh
Comment (thockin) [inline infra/gcp/ensure-prod-storage.sh:364]: Is this sufficient?  Don't we also want to import into BQ ?

**Review by thockin — COMMENTED** [review]
No objections to this alone, but I suspect it will just be the start

File: infra/gcp/ensure-prod-storage.sh
Comment (ameukam) [reply infra/gcp/ensure-prod-storage.sh:364]: @thockin I prefer to leave them the choice of the analysis tool. I would take care of it if there is a need to use BQ.

**Comment by hh** [comment]
@ii will need bq.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
I'll merge and deploy this as-is

Can migrate to a different bucket as followup, which I will track under the umbrella issue for GCS access logs: https://github.com/kubernetes/k8s.io/issues/904

**Comment by spiffxp** [comment]
Ran `./infra/gcp/ensure-prod-storage.sh` but it stopped with this error before getting to the new binding

```  ERROR: (gcloud.iam.service-accounts.add-iam-policy-binding) NOT_FOUND: Not found; Gaia id not found for email k8s-infra-gcr-auditor@k8s-artifacts-prod.iam.gserviceaccount.com```

**Comment by spiffxp** [comment]
OK this is https://github.com/kubernetes/k8s.io/issues/1730 rearing its head.  Time to tackle that to unblock us here

**Comment by spiffxp** [comment]
A rerun appears to have successfully deployed these changes: https://github.com/kubernetes/k8s.io/pull/1998#issuecomment-832256462

**Pull request merged**
